### ISSUE TITLE
I dont know why after install of mailanalyzer, glpi sends white screens

### ISSUE THREAD

**Issue by DulAltamirano** [tomolimo/mailanalyzer#55]
Hello @tomolimo,

I installed the plugin, but do you know why after to install mailanalyzer in glpi 9.5, glpi send white screen when i want to start sesion, i dont know if is by the cookies, or the plugin.

![login](https://user-images.githubusercontent.com/79164411/109694463-74ac9680-7b50-11eb-93ca-f11158824b62.png)

I wish your answer soon.

Thanks, as always.

**Comment by tomolimo** [comment]
Hello @DulAltamirano 
Could you check if you have error messages in the PHP log files?
Thank you
Regards,
Tomolimo

**Comment by DulAltamirano** [comment]
Hello @tomolimo,

error_log

[Wed Mar 03 07:23:20.874393 2021] [php7:notice] [pid 5040] [client ip:50345] PHP Notice:  Undefined property: Auth::$password_expired in /srv/www/htdocs/glpi/inc/session.class.php on line 145, referer: http://ip/glpi/index.php?redirect=%2Ffront%2Fticket.form.php%3Fid%3D138&error=3
[Wed Mar 03 08:12:00.479962 2021] [php7:notice] [pid 26749] [client ip:56559] PHP Notice:  Undefined property: Auth::$password_expired in /srv/www/htdocs/glpi/inc/session.class.php on line 145, referer: http://ip/glpi/index.php?error=1
[Wed Mar 03 09:25:43.980217 2021] [php7:notice] [pid 3068] [client ip:57323] PHP Notice:  Undefined property: Auth::$password_expired in /srv/www/htdocs/glpi/inc/session.class.php on line 145, referer: http://ip/glpi/index.php?error=1

![image](https://user-images.githubusercontent.com/79164411/109830999-a54e0800-7c04-11eb-81cf-cd10b0269628.png)

I dont know if the plugin make that sesion expire but i find this error to.

glpiphplog.CRITICAL:   *** Uncaught Exception Error: Call to undefined method GLPIKey::decryptUsingLegacyKey() in /srv/www/htdocs/ticket/glpi/inc/toolbox.class.php at line 333

Thanks

**Comment by tomolimo** [comment]
Hello @DulAltamirano 

In the logs you showed, there is nothing that concerns the mailanalyzer plugin...
I'm not sure that this issue is linked to mailanalyzer plugin.

Could you post the GLPI configuration?
It's in the Setup > General > System:
![image](https://user-images.githubusercontent.com/7569958/104732694-e3b76280-573d-11eb-8063-20018da5e2f6.png)
Could you copy/paste text from [code] tag to [/code] tag?

Thank you
Regards,
Tomolimo

**Comment by DulAltamirano** [comment]
Hello @tomolimo,

thanks foy your anwer....

I attach the images, i was thinking that was the sesion expired by the plugin because after this started the problem, but i see that is not.
![image](https://user-images.githubusercontent.com/79164411/109983397-65505900-7cc8-11eb-8ee9-28b270c0ad0b.png)
![image](https://user-images.githubusercontent.com/79164411/109984063-09d29b00-7cc9-11eb-8a0d-2ca9223db0ad.png)
![image](https://user-images.githubusercontent.com/79164411/109984118-1525c680-7cc9-11eb-810d-2940f998a6a1.png)
![image](https://user-images.githubusercontent.com/79164411/109984447-66ce5100-7cc9-11eb-8b4d-6f8de4b26153.png)
![image](https://user-images.githubusercontent.com/79164411/109984646-98471c80-7cc9-11eb-876b-d110864f7f1c.png)

Regards.

**Comment by tomolimo** [comment]
Hello @DulAltamirano 

I can see that you are using GLPI 9.5.1, and really I can't support this version of GLPI. You should use the 9.5.4
Thank you to upgrade GLPI

Regards,
Tomolimo

**Comment by DulAltamirano** [comment]
Thanks @tomolimo, but i don't understand.... not exist a solution in this version? :/
In your experience what can be?, if i cant  use 9.5.4 becouse my bd and glpi is personalized.

regards

**Comment by tomolimo** [comment]
Hello @DulAltamirano,
In GLPI 9.5.0 to 9.5.3 there is a bug that prevents the plugin to work. For this bug, I've posted to the GLPI core repository a fix that has been merged into the 9.5.4 version. And to be compatible, I've done a backport of the fix to the 9.5.3 version (as we are running this version).

The only thing I can do for you is to provide you with the patches that must be applied to GLPI, and at your own risk you can try to apply these to your 9.5.1 release.

1. in `toolbox.class.php` in function `addslashes_deep()` add like, shown below, the  `|| is_object($value)`:
![image](https://user-images.githubusercontent.com/7569958/110081087-b2e1ca00-7d8b-11eb-8014-7a61f3283c93.png)

2. in `toolbox.class.php` in function `stripslashes_deep()` add like, shown below, the  `|| is_object($value)`:
![image](https://user-images.githubusercontent.com/7569958/110081164-cf7e0200-7d8b-11eb-841b-ede3d50a0be0.png)

This should be enough for the plugin to run correctly.

Thank you,
Regards,
Tomolimo

**Comment by tomolimo** [comment]
any up?

**Comment by DulAltamirano** [comment]
Hello @tomolimo 

all that you said i did but the issue remains....

i think is the sesion time

**Comment by tomolimo** [comment]
Hello @DulAltamirano 
Could you also a test with only the mailanalyzer plugin being enabled?
We may have to face a conflict of between plugins.
I'm going to install a local release of the GLPI 9.5.1 to try to reproduce your issue.
Thank you
Regards,
Tomolimo

**Comment by tomolimo** [comment]
Hello @DulAltamirano 
I've installed a GLPI 9.5.1 release and I've installed the mailanalyzer plugin.
Find below the source files that need to be patched.
1. in /plugins/mailanalyzer/ folder patch the setup.php file
[setup.php.zip](https://github.com/tomolimo/mailanalyzer/files/6116716/setup.php.zip)

2. in /inc/ folder patch the toolbox.class.php file with the following 
[toolbox.class.php.zip](https://github.com/tomolimo/mailanalyzer/files/6116710/toolbox.class.php.zip)

And I've no troubles with sessions or else...

I suggest that, before patching the files like explained above, you re-install the toolbox.class.php and the mailcollector.class.php files from the official GLPI 9.5.1 archive.

Thank you,
Regards,
Tomolimo
